### PR TITLE
Implement displaying popular instruments list and UI selection

### PR DESCRIPTION
**Pull request by stulees**
- Populates based on category, which is temporarily hardcoded to _currencies_ until we have a HTML template/product that surfaces _category_ attribute. 
- Multi-selection implemented despite there being a future task for this. 
- UI styling applied for selecting an instrument
- When navigating to instrument selector view, any previous selections are cleared. 
- Adding instruments not implemented, this will be implemented when upon completion of Instruments List and Searching instruments

You can validate at https://apps-stage-8.risevision.com/templates/edit/69421d64-e0a2-48f5-95df-923d501c42f8?cid=5ed8d6e8-7eac-4e88-840b-c067d16862a5

cc: @Rise-Vision/apps

### CODE REVIEW

File: web/scripts/template-editor/components/directives/dtv-component-financial.js [hunk 1]
```diff
@@ -16,6 +16,9 @@ angular.module('risevision.template-editor.directives')
             $scope.exitingInstrumentSelector = false;
             $scope.enteringSymbolSelector = false;
             $scope.exitingSymbolSelector = false;
+
+            // TODO: hardcoding category for now until templates have component surface category attribute
+            $scope.category = "currencies";
```
Comment (santiagonoguez) [inline web/scripts/template-editor/components/directives/dtv-component-financial.js]: before closing this card you could add this attribute manually to the blueprint, and then use the function in my latest PR to get that value

https://github.com/Rise-Vision/rise-vision-apps/pull/849

File: web/partials/template-editor/components/component-financial.html [hunk 2]
```diff
@@ -46,16 +93,22 @@ <h2 translate>template.financial.no-results.title</h2>
     </div>
   </div>
 
-  <div class="instrument-selector-popular" translate>template.financial.most-popular-category.currencies</div>
+  <div class="instrument-selector-popular" ng-show="instrumentSearch.length > 0" translate>template.financial.most-popular-category.currencies</div>
```
Comment (santiagonoguez) [inline web/partials/template-editor/components/component-financial.html]: the label should be derived from the category in the scope

File: web/partials/template-editor/components/component-financial.html [hunk 3]
```diff
@@ -25,15 +72,15 @@
     ng-show="showSymbolSelector || enteringSymbolSelector">
   <div class="input-group">
     <span class="input-group-addon"><i class="fa fa-search"></i></span>
-    <input type="text" class="form-control"
+    <input type="text" class="form-control" ng-model="searchKeyword"
```
Comment (santiagonoguez) [inline web/partials/template-editor/components/component-financial.html]: the placeholder and label below should be derived from the category in scope

Comment (stulees) [reply web/partials/template-editor/components/component-financial.html]: @santiagonoguez This PR is not for searching instruments, those aspects will come on my next card. I had to add a model to the input for now to facilitate which financial request to make. Thanks

File: web/partials/template-editor/components/component-financial.html (hunk 2 above)
Comment (stulees) [reply web/partials/template-editor/components/component-financial.html]: @santiagonoguez Ah yes, I'll change that, thanks

File: web/scripts/template-editor/components/directives/dtv-component-financial.js (hunk 1 above)
Comment (stulees) [reply web/scripts/template-editor/components/directives/dtv-component-financial.js]: @santiagonoguez Good idea on adding manually to blueprint, I will do that. I'd prefer to close this card out though as I can use your function in my next card for searching, which touches functionality here anyway. Thoughts?

Comment (santiagonoguez) [reply web/scripts/template-editor/components/directives/dtv-component-financial.js]: I agree, so this detail doesn't stop you for the moment.

**Comment by stulees** [comment]
@santiagonoguez Once I get approval from Matt I will be adding the inline styles to common-header and will update this PR

File: web/partials/template-editor/components/component-financial.html (hunk 3 above)
Comment (santiagonoguez) [reply web/partials/template-editor/components/component-financial.html]: +1

File: web/scripts/template-editor/components/directives/dtv-component-financial.js [hunk 4]
```diff
@@ -77,6 +77,10 @@ angular.module('risevision.template-editor.directives')
               } );
           };
 
+          $scope.getPopularTitle = function() {
+            return 'template.financial.most-popular-category.' + $scope.category;
```
Comment (santiagonoguez) [inline web/scripts/template-editor/components/directives/dtv-component-financial.js]: @stulees later we'll need to remember when obtaining the category and assigning it to $scope.category, is to translate spaces into dashes, so this doesn't break, for example 'world indexes' category -> 'world-indexes'.

Comment (stulees) [reply web/scripts/template-editor/components/directives/dtv-component-financial.js]: @santiagonoguez yes and noted, will take care of that in Search card, thanks!

File: web/partials/template-editor/components/component-financial.html (hunk 2 above)
Comment (stulees) [reply web/partials/template-editor/components/component-financial.html]: @santiagonoguez Corrected, thanks

File: web/scripts/template-editor/components/directives/dtv-component-financial.js [hunk 5]
```diff
@@ -51,6 +54,33 @@ angular.module('risevision.template-editor.directives')
...
+              instrumentSearchService.keywordSearch( $scope.category, $scope.searchKeyword ) :
+              instrumentSearchService.popularSearch( $scope.category );
+
+            $scope.searching = true;
+            promise.then( function( res ) {
+              $scope.instrumentSearch = angular.copy( res );
+              $scope.searching = false;
+            } )
+              .catch( function( err ) {
+                console.error( err );
```
Comment (santiagonoguez) [inline web/scripts/template-editor/components/directives/dtv-component-financial.js]: @stulees @fjvallarino I've seen both $log.error and console.error being used all along Apps scripts, is there some standard or consensus or we shouldn't care ?

Comment (stulees) [reply web/scripts/template-editor/components/directives/dtv-component-financial.js]: @santiagonoguez I'm not sure, @fjvallarino any thoughts?

Comment (fjvallarino) [reply web/scripts/template-editor/components/directives/dtv-component-financial.js]: @santiagonoguez @stulees in general `$log` is preferred

**Comment by stulees** [comment]
@santiagonoguez @fjvallarino I've removed inline styles and brought them in via bump of common-header version. Please review, thanks

File: web/scripts/template-editor/components/directives/dtv-component-financial.js (hunk 5 above)
Comment (stulees) [reply web/scripts/template-editor/components/directives/dtv-component-financial.js]: @fjvallarino @santiagonoguez Okay, in my next PR(s) for Search functionality I will ensure to correct this, and presume we all use `$log` going forward, thanks

**Pull request merged**